mitreattack/navlayers/layerGenerator_cli.py

```diff
@@ -0,0 +1,45 @@
+import argparse
+
+try:
+    from .generators.overview_generator import OverviewGenerator
+    from .generators.usage_generator import UsageGenerator
+except ImportError:
+    from mitreattack.navlayers.generators.overview_generator import OverviewGenerator
+    from mitreattack.navlayers.generators.usage_generator import UsageGenerator
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Generate an ATT&CK Navigator layer')
+    parser.add_argument('-m', '--mode', choices=['overview', 'usage'], required=True,
+                        help='Generate a complete overview or the specific usage for a entity')
+    parser.add_argument('-t', '--type', choices=['group', 'software', 'mitigation'], required=True,
+                        help='What kind of object to build the layer for')
+    parser.add_argument('-i', '--identity', help='The identity (id/name) of the desired entity '
+                                                 '(required for usage generation)', default=None)
```

Comment:
Identity is not the correct word here especially since `identity` is a STIX type. Probably this would be better?

- `--mapped-to`
- `Output techniques mapped to the given, group, software, or mitigation. Argument can be name, associated group/software, or ATT&CK ID.`

Reply:
For clarity I think we should redo the arguments completely:
- `--overview [type]` would output the overview layer for the given type
- `--mapped-to [pattern]` would do the usage layer. With refactoring (mentioned elsewhere in the review) I don't _think_ we need to explicitly know the type for this option since only one object of any type should match the pattern.

Reply:
If arguments are redone in this way, use a mutually exclusive argument group to ensure only one of the two options is given: https://docs.python.org/3/library/argparse.html#mutual-exclusion

Reply:
I can get behind that. I've made the appropriate changes.